### PR TITLE
use php_setcookie function (PHP public API)

### PR DESCRIPTION
**Pull request by remicollet**
Simplify a lot the function (using PHP public API instead of `call_user_function` hack)

This make the samesite test passes on 7.1, 7.2 and 7.3

I think it can be a nice idea to handle this to allow user to have same configuration on various PHP versions.

### CODE REVIEW

**Comment by remicollet** [comment]
Notice: I add a TODO about the RETURN_TRUE, because (looking at php sources), should be false if php_setcookie fails (but this will change the test expected output)

**Comment by xXx-caillou-xXx** [comment]
I have no idea why we always `RETURN_TRUE`, you can modify it if you want.

File: src/sp_cookie_encryption.c [hunk 1]
```diff
@@ -42,13 +43,14 @@ static zend_string *encrypt_data(zend_string *data) {
 }
 
 PHP_FUNCTION(sp_setcookie) {
-  zval params[7] = {{{0}}};
   zend_string *name = NULL, *value = NULL, *path = NULL, *domain = NULL,
-              *samesite = NULL;
+#if PHP_VERSION_ID >= 70300
+              *samesite = NULL,
+#endif
+              *value2 = NULL, *path2 = NULL;
```
Comment (xXx-caillou-xXx) [inline src/sp_cookie_encryption.c]: Maybe a better naming scheme, like `value_enc` ?
`path2` could only be used if `PHP_VERSION_ID` < 70300, and maybe rename it to `path_samesite` or something like that?

Comment (remicollet) [reply src/sp_cookie_encryption.c]: done

File: src/sp_cookie_encryption.c [hunk 2]
```diff
@@ -88,64 +92,54 @@ PHP_FUNCTION(sp_setcookie) {
...
+
+    samesite = zend_string_init(cookie_samesite, strlen(cookie_samesite), 0);
+#endif
+  }
+
+
+#if PHP_VERSION_ID < 70300
+  if (php_setcookie(name, (value_enc ? value_enc : value), expires, (path_samesite ? path_samesite : path), domain, secure, 1, httponly)) {
+#else
+  if (php_setcookie(name, (value_enc ? value_enc : value), expires, path, domain, secure, httponly, samesite, 1)) {
```
Comment (xXx-caillou-xXx) [inline src/sp_cookie_encryption.c]: What if another module wants to hook `setcookie`?

Comment (remicollet) [reply src/sp_cookie_encryption.c]: indeed.. will be an issue (depending of load order)

Comment (jvoisin) [reply src/sp_cookie_encryption.c]: I'm not sure that we want to prevent other extension hooking the function from working :/

**Comment by jvoisin** [comment]
I've documented this behaviour in 2ad1b9ab2efefd063d9f412e56d6a642187918a1

**Pull request merged**